automation/test.sh

```diff
@@ -170,143 +40,111 @@ for filename in dist/templates/*; do
-    sleep $sample
-
-    current_time=$((current_time + sample))
-    if [ $current_time -gt $timeout ]; then
-      exit 1
-    fi
-  done
-  _oc get pods -n $i
-done
-
-# Used to store the exit code of the webhook creation command
-webhookUpdated=1
-webhookUpdateRetries=10
-
-while [ $webhookUpdated != 0 ];
-do
-  # Approve all CSRs to avoid an issue similar to this: https://github.com/kubernetes/frakti/issues/200
-  # This is attempted before any call to 'oc exec' because there's suspect that more csr's have to be approved
-  # over time and not only once after the cluster has been initiated.
-  _oc adm certificate approve $(_oc get csr -ocustom-columns=NAME:metadata.name --no-headers)
-
-  if [ $webhookUpdateRetries == 0 ];
-  then
-    echo Retry count for template validator ca bundle injection reached
-    exit 1
-  fi
-
-  webhookUpdateRetries=$((webhookUpdateRetries-1))
-
-  VALIDATOR_POD=$(_oc get pod -n $NAMESPACE -l kubevirt.io=virt-template-validator -o json | jq -r .items[0].metadata.name)
-  if [ "$VALIDATOR_POD" == "" ];
-  then
-    # Retry if an error occured
-    continue
-  fi
+oc wait --for=condition=Available --timeout=${timeout}s kubevirt/kubevirt -n $namespace
+
+oc apply -f - <<EOF
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kubevirt-config
+  namespace: kubevirt
+data:
+  feature-gates: "DataVolumes"
+---
+EOF
+
+key="/tmp/secrets/accessKeyId"
+token="/tmp/secrets/secretKey"
+
+if test -f "$key" && test -f "$token"; then
+  id=$(cat $key | tr -d '\n' | base64)
+  token=$(cat $token | tr -d '\n' | base64 | tr -d ' \n')
+
+  oc apply -n $namespace -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: common-templates-container-disk-puller
+  labels:
+    app: containerized-data-importer
+type: Opaque
+data:
+  accessKeyId: "${id}"
+  secretKey: "${token}"
+EOF
+fi
 
-  CA_BUNDLE=$(_oc exec -n $NAMESPACE $VALIDATOR_POD -- /bin/cat /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt | base64 -w 0)
-  if [ "$CA_BUNDLE" == "" ];
-  then
-    # Retry if an error occured
-    continue
+echo "Deploying CDI"
+export CDI_VERSION=$(curl -s https://api.github.com/repos/kubevirt/containerized-data-importer/releases | \
+            jq '.[] | select(.prerelease==false) | .tag_name' | sort -V | tail -n1 | tr -d '"')
+oc apply -f https://github.com/kubevirt/containerized-data-importer/releases/download/$CDI_VERSION/cdi-operator.yaml
+oc apply -f https://github.com/kubevirt/containerized-data-importer/releases/download/$CDI_VERSION/cdi-cr.yaml
+oc rollout status -n cdi deployment/cdi-operator
+oc apply -f - <<EOF
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: cdi-role
+  namespace: cdi
+rules:
+- apiGroups: ["cdi.kubevirt.io"]
+  resources: ["datavolumes/source"]
+  verbs: ["*"]
+---
+EOF
+
+export SSP_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt-ssp-operator/releases | \
```

Comment:
I would prefer deploying template validator without the operator, it allows us to consume released versions before the reach the operator

Reply:
I am preferring deployment with ssp operator because:
1) we don't have to deal with certificates (sometimes whole test failed due to certificate issue). 
2) Templates will be tested with validator which is shipped with HCO
3) validator deploy manifests in original repo are not prepared to be deployed on OCP 4 yet

Reply:
For the next release, we have changes for both validator and templates, if we use the operator here, it means the operator has to consume the new TV before it was tested with templates, which is not a good idea.
IMO the proper way would be to test templates with validator first, and then let the operator consume them